src/ssl/context.cpp

```diff
@@ -0,0 +1,287 @@
+/*
+
+Copyright (c) 2020, Paul-Louis Ageneau
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+  * Redistributions of source code must retain the above copyright
+    notice, this list of conditions and the following disclaimer.
+  * Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimer in
+    the documentation and/or other materials provided with the distribution.
+  * Neither the name of the author nor the names of its
+    contributors may be used to endorse or promote products derived
+    from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifdef TORRENT_USE_SSL
+
+#include "libtorrent/ssl/context.hpp"
+#include "libtorrent/ssl/stream.hpp"
+#include "libtorrent/settings_pack.hpp"
+
+#define DEBUG_SSL_VERIFY 0
+
+#if DEBUG_SSL_VERIFY
+#include <cstdarg> // for va_list
+#include <cstdio> // for vsnprintf
+#endif
+
+namespace {
+
+#if DEBUG_SSL_VERIFY
+void debug_log(char const* fmt, ...)
```

Comment:
this used to print logsas alerts, and make them available to clients. I see the verification functions used to be member functions on `torrent`, was the alert the only thing it needed the torrent object for?

Reply:
Yes it was the only thing. So the alerts were only available in debug mode by design?

Reply:
it was enabled unless `TORRENT_DISABLE_LOGGING` was defined, which actually makes it available by default (but only when enabled in the `alert_mask`). I don't really have strong opinions about preserving this logging though. I imagine I added it while trouble shooting certs issues in SSL swarms